Issue with CMake when installing

When installing I get this error from CMake
```
  CMake Warning (dev) at /tmp/pip-build-env-ttsh28xu/overlay/lib/python3.13/site-packages/cmake/data/share/cmake-4.0/Modules/FetchContent.cmake:1953 (message):
    Calling FetchContent_Populate(yaml-cpp) is deprecated, call
    FetchContent_MakeAvailable(yaml-cpp) instead.  Policy CMP0169 can be set to
    OLD to allow FetchContent_Populate(yaml-cpp) to be called directly for now,
    but the ability to call it with declared details will be removed completely
    in a future version.
  Call Stack (most recent call first):
    CMakeLists.txt:19 (FetchContent_Populate)
  This warning is for project developers.  Use -Wno-dev to suppress it.

  CMake Error at /tmp/pip-install-ryajgtk5/wuff_1b70ed721a6b4d6da9ad4fa34b9c801c/build/temp.linux-x86_64-cpython-313/wuff/_deps/yaml-cpp-src/CMakeLists.txt:2 (cmake_minimum_required):
    Compatibility with CMake < 3.5 has been removed from CMake.

    Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
    to tell CMake that the project requires at least <min> but has been updated
    to work with policies introduced by <max> or earlier.

    Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,41 +1,38 @@
-cmake_minimum_required(VERSION 3.12)
+cmake_minimum_required(VERSION 3.5)
 project(wuff)
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-# - - - YAML-CPP
 include(FetchContent)
 
+# - - - YAML-CPP
 FetchContent_Declare(
-        yaml-cpp
-        GIT_REPOSITORY https://github.com/jbeder/yaml-cpp.git
-        GIT_TAG 0.8.0
+    yaml-cpp
+    GIT_REPOSITORY https://github.com/jbeder/yaml-cpp.git
+    GIT_TAG master
 )
-FetchContent_GetProperties(yaml-cpp)
-
-if (NOT yaml-cpp_POPULATED)
-    message(STATUS "Fetching yaml-cpp...")
-    FetchContent_Populate(yaml-cpp)
-    add_subdirectory(${yaml-cpp_SOURCE_DIR} ${yaml-cpp_BINARY_DIR})
-endif ()
+FetchContent_MakeAvailable(yaml-cpp)
 
 # - - - Tree-sitter
 FetchContent_Declare(
-        tree_sitter
-        GIT_REPOSITORY https://github.com/tree-sitter/tree-sitter.git
-        GIT_TAG v0.20.9
+    tree_sitter
+    GIT_REPOSITORY https://github.com/tree-sitter/tree-sitter.git
+    GIT_TAG v0.20.9
 )
 FetchContent_GetProperties(tree_sitter)
-
 if (NOT tree_sitter_POPULATED)
     message(STATUS "Fetching tree-sitter...")
     FetchContent_Populate(tree_sitter)
+
     # Since Tree-sitter does not use CMake, we manually add its source and include directories
     set(TREE_SITTER_SRC ${tree_sitter_SOURCE_DIR}/lib/src/lib.c)
-    set(TREE_SITTER_INCLUDE_DIRS ${tree_sitter_SOURCE_DIR}/lib/include ${tree_sitter_SOURCE_DIR}/lib/src ${tree_sitter_SOURCE_DIR}/lib/src/unicode)
-    message(STATUS "Include directories: ${TREE_SITTER_INCLUDE_DIRS}")
-endif ()
+    set(TREE_SITTER_INCLUDE_DIRS
+        ${tree_sitter_SOURCE_DIR}/lib/include
+        ${tree_sitter_SOURCE_DIR}/lib/src
+        ${tree_sitter_SOURCE_DIR}/lib/src/unicode
+    )
+endif()
 
 # Tree-Sitter WooWoo
 set(WOOWOO_PARSER_SRC tree-sitter/woowoo/parser.c)
@@ -48,24 +45,68 @@ set(YAML_SCANNER_SRC tree-sitter/yaml/scanner.cc)
 # Tree-Sitter BibTeX
 set(BIBTEX_PARSER_SRC tree-sitter/bibtex/parser.c)
 
-# - - - - -
-
+# - - - Pybind11
 FetchContent_Declare(
-        pybind11
-        GIT_REPOSITORY https://github.com/pybind/pybind11.git
-        GIT_TAG        v2.11.1
+    pybind11
+    GIT_REPOSITORY https://github.com/pybind/pybind11.git
+    GIT_TAG v2.11.1
 )
-
-# Make pybind11 available for use
 FetchContent_MakeAvailable(pybind11)
+
+# - - - Build main module
 pybind11_add_module(${PROJECT_NAME} NO_EXTRAS
-        ${TREE_SITTER_SRC} # Add Tree-sitter lib.c source
+    ${TREE_SITTER_SRC}
+    ${WOOWOO_PARSER_SRC}
+    ${WOOWOO_SCANNER_SRC}
+    ${YAML_PARSER_SRC}
+    ${YAML_SCANNER_SRC}
+    ${BIBTEX_PARSER_SRC}
+    Bindings.cpp
+    WooWooAnalyzer.cpp
+    project/WooWooDocument.cpp
+    project/WooWooProject.cpp
+    project/DialectedWooWooDocument.cpp
+    project/MetaContext.cpp
+    project/CommentLine.cpp
+    project/UTF8toUTF16Mapping.cpp
+    project/Woofile.cpp
+    dialect/DialectManager.cpp
+    dialect/DocumentPart.cpp
+    dialect/Dialect.cpp
+    dialect/MetaBlock.cpp
+    dialect/Field.cpp
+    dialect/Shorthand.cpp
+    dialect/Reference.cpp
+    dialect/Environment.cpp
+    dialect/Wobject.cpp
+    components/Component.cpp
+    components/Hoverer.cpp
+    components/Highlighter.cpp
+    components/Navigator.cpp
+    components/Completer.cpp
+    components/Linter.cpp
+    components/Folder.cpp
+    parser/Parser.cpp
+    utils/utils.cpp
+)
+target_include_directories(${PROJECT_NAME} SYSTEM PRIVATE ${TREE_SITTER_INCLUDE_DIRS})
+target_link_libraries(${PROJECT_NAME} PUBLIC yaml-cpp::yaml-cpp)
+
+if (APPLE)
+    target_compile_options(${PROJECT_NAME} PRIVATE "-mmacosx-version-min=10.15")
+    target_link_options(${PROJECT_NAME} PRIVATE "-mmacosx-version-min=10.15")
+endif()
+
+# - - - WooWooTest executable
+if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/main.cpp")
+    add_executable(WooWooTest
+        ${TREE_SITTER_SRC}
         ${WOOWOO_PARSER_SRC}
         ${WOOWOO_SCANNER_SRC}
         ${YAML_PARSER_SRC}
         ${YAML_SCANNER_SRC}
         ${BIBTEX_PARSER_SRC}
-        Bindings.cpp
+        main.cpp
         WooWooAnalyzer.cpp
         project/WooWooDocument.cpp
         project/WooWooProject.cpp
@@ -92,56 +133,14 @@ pybind11_add_module(${PROJECT_NAME} NO_EXTRAS
         components/Folder.cpp
         parser/Parser.cpp
         utils/utils.cpp
-)
-target_include_directories(${PROJECT_NAME} SYSTEM PRIVATE ${TREE_SITTER_INCLUDE_DIRS})
-target_link_libraries(${PROJECT_NAME} PUBLIC yaml-cpp::yaml-cpp)
-if(APPLE)
-    target_compile_options(${PROJECT_NAME} PRIVATE "-mmacosx-version-min=10.15")
-    target_link_options(${PROJECT_NAME} PRIVATE "-mmacosx-version-min=10.15")
-endif()
-    
-# - - - WooWooTest
-if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/main.cpp")
-    add_executable(WooWooTest
-            ${TREE_SITTER_SRC}
-            ${WOOWOO_PARSER_SRC}
-            ${WOOWOO_SCANNER_SRC}
-            ${YAML_PARSER_SRC}
-            ${YAML_SCANNER_SRC}
-            ${BIBTEX_PARSER_SRC}
-            main.cpp
-            WooWooAnalyzer.cpp
-            project/WooWooDocument.cpp
-            project/WooWooProject.cpp
-            project/DialectedWooWooDocument.cpp
-            project/MetaContext.cpp
-            project/CommentLine.cpp
-            project/UTF8toUTF16Mapping.cpp
-            project/Woofile.cpp
-            dialect/DialectManager.cpp
-            dialect/DocumentPart.cpp
-            dialect/Dialect.cpp
-            dialect/MetaBlock.cpp
-            dialect/Field.cpp
-            dialect/Shorthand.cpp
-            dialect/Reference.cpp
-            dialect/Environment.cpp
-            dialect/Wobject.cpp
-            components/Component.cpp
-            components/Hoverer.cpp
-            components/Highlighter.cpp
-            components/Navigator.cpp
-            components/Completer.cpp
-            components/Linter.cpp
-            components/Folder.cpp
-            parser/Parser.cpp
-            utils/utils.cpp
     )
     target_include_directories(WooWooTest SYSTEM PRIVATE ${TREE_SITTER_INCLUDE_DIRS})
     target_link_libraries(WooWooTest PUBLIC yaml-cpp::yaml-cpp pybind11::embed)
-    if(MSVC)
+
+    if (MSVC)
         target_compile_options(WooWooTest PRIVATE /W4)
     else()
         target_compile_options(WooWooTest PRIVATE -Wall -Wextra -Wpedantic)
     endif()
-endif ()
+endif()
+

